.buildkite/scripts/package.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common.sh
+
+VERSION=$(awk '/const DefaultVersion/{print $NF}' version/version.go | tr -d '"')
+PLATFORM_TYPE=$(uname -m)
+MATRIX_TYPE="$1"
+INFRA_REPO="https://github.com/repos/elastic/infra/contents"
+
+if [[ ${BUILDKITE_BRANCH} == "main" && ${MATRIX_TYPE} == "staging" ]]; then
+    echo "INFO: staging artifacts for the main branch are not required."
+    exit 0
+else
+    PLATFORMS=""
+    PACKAGES=""
+    if [[ ${PLATFORM_TYPE} == "arm" || ${PLATFORM_TYPE} == "aarch64" ]]; then
+        PLATFORMS="linux/arm64"
+        PACKAGES="docker"
+    fi
+
+    add_bin_path
+    with_go
+    with_mage
+
+    if [[ ${MATRIX_TYPE} == "staging" ]]; then
+        make release
+    else
+        make SNAPSHOT=true release
+    fi
+    google_cloud_auth
+    upload_mbp_packages_to_gcp_bucket "build/distributions/" "${MATRIX_TYPE}"
+fi
```

Comment:
No need of the `else` block since if it is true, it already runs `exit 0`

```suggestion
if [[ ${BUILDKITE_BRANCH} == "main" && ${MATRIX_TYPE} == "staging" ]]; then
    echo "INFO: staging artifacts for the main branch are not required."
    exit 0
PLATFORMS=""
PACKAGES=""
if [[ ${PLATFORM_TYPE} == "arm" || ${PLATFORM_TYPE} == "aarch64" ]]; then
    PLATFORMS="linux/arm64"
    PACKAGES="docker"
fi

add_bin_path
with_go
with_mage

if [[ ${MATRIX_TYPE} == "staging" ]]; then
    make release
else
    make SNAPSHOT=true release
fi

google_cloud_auth
upload_mbp_packages_to_gcp_bucket "build/distributions/" "${MATRIX_TYPE}"
```